Source/AppAuth.h

```diff
@@ -41,6 +41,8 @@
 #import "OIDTokenUtilities.h"
 #import "OIDURLSessionProvider.h"
 
+#ifdef APPAUTH_USERAGENT
```

Comment:
This will break existing users who are not using Pods, needs to be inverted (#ifndef _NO_APPAUTH_USERAGENT and define it in Core).

Reply:
I don't think it'll work by inverting this because Core is now included as a dependency in ExternalUserAgent, so I guess this macro will always be defined... And also if no spec is specified in the user podfile, I think all specs will be included.

If this is confirmed, maybe another solution is to have ExternalUserAgent as a default_subspec and include all the files under Source folder in it.

I'll try to test it.

Reply:
Good points, look forward to hearing the results of your experiment!

Reply:
I confirm that the flag is always defined no matter which subspec is specified in the Podfile. And I also had another problem when importing `pod AppAuth/Core` in a Swift project, #ifndef seems to be ignored and iOS specific files/types are imported...

Reply:
I didn't have much time to test this yet, but I also couldn't make it work with defining `default_subspec` and inverting the preprocessor macro (i.e. using `#ifndef NO_APPAUTH_USERAGENT`) in the header...

But I'm still thinking if it's really a problem to keep the existing logic (i.e. `#ifdef APPAUTH_USERAGENT`).... in my current understanding the `AppAuth.h` is currently used exclusively for the CocoaPods build. Carthage/manual framework builds use another AppAuth.h header. So do we really need to consider users who are not using Pods in this header file?

Reply:
If they're using a different header, then I guess it should be fine to keep the existing logic, assuming this file is only used in the Cocoapods environment, where this will always be defined.

Do all the samples (both Cocoapods and Carthage) compile with this PR as it is currently?

@julienbodet can you review & LGTM when/if you confirm that this PR looks good for your use-cases?

Reply:
I guess I need to do some more testing. Although the examples seem to build just fine I'm having issues (in my own project) that at runtime the user agent extension methods of cannot be found:

```
+[OIDAuthState authStateByPresentingAuthorizationRequest:presentingViewController:callback:]: unrecognized selector sent to class 0x10cf7c9f0
```

Reply:
Please let me know if calling the category functions works on your side .. but I couldn't make it work for me yet. Seems like the `-ObjC` flag isn't working for subpods or something like that.

Reply:
Couple of things: 1) I think the headers should match, ideally, divergence just confuses things – and Carthage users probably need this too! 2) The other way to retain the #if logic is to add that define to the xcodeproj build settings so that Carthage users will have it defined, that way they won't break when they upgrade either.

I've gone an implemented this in #318, and I added a new framework target so Carthage users can build with extensions too.

In my testing it works, note that I defined the variable in `GCC_PREPROCESSOR_DEFINITIONS` rather than `OTHER_CFLAGS` in this PR.